app/services/participants/change_schedule/validate_and_change_schedule.rb

```diff
@@ -26,6 +27,22 @@ def schedule
       def not_already_withdrawn
         errors.add(:participant_id, I18n.t(:withdrawn_participant)) if participant_profile_state&.withdrawn?
       end
+
+      def schedule_valid_with_pending_declarations
+        return unless user_profile
+
+        declarations = user_profile.participant_declarations.not_voided
+        declarations.each do |declaration|
+          milestone = schedule.milestone_for_declaration_type[declaration.declaration_type]
+          unless milestone.start_date.beginning_of_day < declaration.declaration_date
```

Comment:
```suggestion
          if declaration.declaration_date < milestone.start_date.beginning_of_day
```

These read a bit better if keep the target first and don't do the negation.

Reply:
Do we not allow declarations to be made on the milestone date itself? Do they have to be before it?

Reply:
Well, we have tests (that I might have broken locally) that cover all edge cases.